packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -2609,74 +2539,215 @@ export class LitNodeClientNodeJs
     const responseData = res.values;
     logWithRequestId(
       requestId,
-      'responseData',
+      '[signSessionKey] responseData',
       JSON.stringify(responseData, null, 2)
     );
 
     // ========== Extract shares from response data ==========
     // -- 1. combine signed data as a list, and get the signatures from it
-    const signedDataList = responseData.map(
-      (r: any) => (r as SignedData).signedData
-    );
+    let curveType = responseData[0]?.curveType;
+
+    if (!curveType) {
+      log(`[signSessionKey] curveType not found. Defaulting to ECDSA.`);
+      curveType = 'ECDSA';
+    }
 
-    logWithRequestId(requestId, 'signedDataList', signedDataList);
+    log(`[signSessionKey] curveType is "${curveType}"`);
+
+    let signedDataList: any[] = [];
+
+    if (curveType === LIT_CURVE.BLS) {
+      let _responseData: BlsResponseData[] = responseData;
+
+      const signatureShares = _responseData.map((s) => ({
+        ProofOfPossession: s.signatureShare.ProofOfPossession,
+      }));
+
+      log(`[signSessionKey] signatureShares:`, signatureShares);
+
+      signedDataList = _responseData.map((s) => {
+        return s.dataSigned;
+      });
+
+      signedDataList = _responseData;
+    } else {
+      signedDataList = responseData.map(
+        (r: any) => (r as SignedData).signedData
+      );
+    }
+
+    if (signedDataList.length <= 0) {
+      const err = `[signSessionKey] signedDataList is empty.`;
+      log(err);
+      throw new Error(err);
+    }
+
+    logWithRequestId(
+      requestId,
+      '[signSessionKey] signedDataList',
+      signedDataList
+    );
```

Comment:
You're combing the BLS shares into the final signature? Extract this out in a function and add unit tests for it? How can we be sure whether the BLS share combination is correct?

Reply:
That makes sense. Will do that

Reply:
Do we have unit tests for it now?